cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AppSystemMetadataWriter.java

```diff
@@ -72,9 +78,27 @@ public AppSystemMetadataWriter(MetadataServiceClient metadataServiceClient, Appl
         existingPluginClasses.add(plugin.getPluginClass());
       }
     }
+    addAccelerators(appSpec, applicationClass, properties);
     return properties.build();
   }
 
+  private void addAccelerators(ApplicationSpecification appSpec,
+                               ApplicationClass applicationClass,
+                               ImmutableMap.Builder<String, String> properties) {
+    //add from all plugins
+    Set<String> acceleratorSet = appSpec.getPlugins().values().stream().map(Plugin::getPluginClass)
+      .map(PluginClass::getRequirements).map(
+        Requirements::getAccelerators).flatMap(Set::stream)
+      .collect(Collectors.toSet());
+    //add from application
+    acceleratorSet.addAll(applicationClass.getRequirements().getAccelerators());
+    String acceleratorsValue = String.join(",", acceleratorSet);
+    if (acceleratorsValue == null || acceleratorsValue.isEmpty()) {
```

Comment:
It won't be `null`, right?

Reply:
Yes :) . Changed to size check for the set.